cli/src/helpers/dynamic_format.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+use time::format_description::{self, BorrowedFormatItem};
+use tracing::{Event, Subscriber};
+use tracing_subscriber::{
+    fmt::{format::Writer, FmtContext, FormatEvent, FormatFields},
+    registry::LookupSpan,
+};
+
+pub struct DynamicFormatter<'b> {
+    dim_format: DimFormat<'b>,
+    default_format: tracing_subscriber::fmt::format::Format,
+    dim: Arc<AtomicBool>,
+}
+
+impl<'b, S, N> FormatEvent<S, N> for DynamicFormatter<'b>
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    N: for<'a> FormatFields<'a> + 'static,
+{
+    fn format_event(&self, ctx: &FmtContext<'_, S, N>, writer: Writer<'_>, event: &Event<'_>) -> std::fmt::Result {
+        if self.dim.load(Ordering::Relaxed) {
+            self.dim_format.format_event(ctx, writer, event)
+        } else {
+            self.default_format.format_event(ctx, writer, event)
+        }
+    }
+}
+
+impl<'b> DynamicFormatter<'b> {
+    pub fn new(dim: Arc<AtomicBool>) -> Self {
+        let dim_format = DimFormat::new();
+        let default_format = tracing_subscriber::fmt::format::Format::default();
+        Self { dim_format, default_format, dim }
+    }
+}
+
+struct DimFormat<'b> {
+    fmt: Vec<BorrowedFormatItem<'b>>,
+}
+
+impl<'b> DimFormat<'b> {
+    fn new() -> Self {
+        let format = format_description::parse("[year]-[month]-[day]T[hour]:[minute]:[second].[subsecond digits:6]Z")
+            .expect("failed to set timestampt format");
+        Self { fmt: format }
+    }
+}
+
+impl<'b, S, N> FormatEvent<S, N> for DimFormat<'b>
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    N: for<'a> FormatFields<'a> + 'static,
+{
+    fn format_event(&self, ctx: &FmtContext<'_, S, N>, mut writer: Writer<'_>, event: &Event<'_>) -> std::fmt::Result {
```

Comment:
It might be a good idea to explain what this function does (seems to me that this just dims the regular event log display), in case there is some simpler derive macro (or something else) that can do this in the future.